pvlib/tests/test_irradiance.py

```diff
@@ -1082,3 +1083,71 @@ def test_clearness_index_zenith_independent(airmass_kt):
+                      'dni': [49.63565561689957, 62.10624908037814]},
+                     index=times)
+    # Define the solar position and clearsky dataframes
+    solar_position = pd.DataFrame({'apparent_zenith': [71.7303262449161,
+                                                       59.369],
+                                   'zenith': [71.7764, 59.395]},
+                                  index=pd.DatetimeIndex([
+                                      '2010-07-05 07:00:00-0700',
+                                      '2010-07-05 08:00:00-0700']))
+    clearsky = pd.DataFrame({'dni': [625.5254880160008, 778.7766443075865],
+                             'ghi': [246.3508023804681, 469.461381740857],
+                             'dhi': [50.25488725346631, 72.66909939636372]},
+                            index=pd.DatetimeIndex([
+                                '2010-07-05 07:00:00-0700',
+                                '2010-07-05 08:00:00-0700']))
+    # Test scenario where DNI is generated via component sum equation
+    complete_df = irradiance.complete_irradiance(
+        solar_position.apparent_zenith,
+        ghi=i.ghi,
+        dhi=i.dhi,
+        dni=None,
+        dni_clear=clearsky.dni)
+    # Assert that the ghi, dhi, and dni series match the original dataframe
+    # values
+    assert_frame_equal(complete_df, i)
+    # Test scenario where GHI is generated via component sum equation
+    complete_df = irradiance.complete_irradiance(
+        solar_position.apparent_zenith,
+        ghi=None,
+        dhi=i.dhi,
+        dni=i.dni,
+        dni_clear=clearsky.dni)
+    # Assert that the ghi, dhi, and dni series match the original dataframe
+    # values
+    assert_frame_equal(complete_df, i)
+    # Test scenario where DHI is generated via component sum equation
+    complete_df = irradiance.complete_irradiance(
+        solar_position.apparent_zenith,
+        ghi=i.ghi,
+        dhi=None,
+        dni=i.dni,
+        dni_clear=clearsky.dni)
+    # Assert that the ghi, dhi, and dni series match the original dataframe
+    # values
+    assert_frame_equal(complete_df, i)
+    # Test scenario where all parameters are passed (throw error)
+    complete_df = assert_raises(ValueError,
+                                irradiance.complete_irradiance,
+                                solar_position.apparent_zenith,
+                                ghi=i.ghi,
+                                dhi=i.dhi,
+                                dni=i.dni,
+                                dni_clear=clearsky.dni)
+    # Test scenario where only one parameter is passed (throw error)
+    complete_df = assert_raises(ValueError,
+                                irradiance.complete_irradiance,
+                                solar_position.apparent_zenith,
+                                ghi=None,
+                                dhi=None,
+                                dni=i.dni,
+                                dni_clear=clearsky.dni)
```

Comment:
for consistency with existing test pattern, I would rather use [`with pytest.raises(ValueError)`](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions). Is there a reason to prefer numpy's `assert_raises`? I don't recall seeing that until now.

Reply:
Changed this to pytest.raises in commit https://github.com/pvlib/pvlib-python/pull/1567/commits/99205b4daa88910c3c3e3d0ee592799ac3605404